_maps/__MAP_DEFINES.dm

```diff
@@ -28,4 +52,19 @@ Last space-z level = empty
 #define EMPTY_AREA_8 "Empty Area 8"
 #define AWAY_MISSION "Away Mission"
 #define AWAY_MISSION_LIST list(AWAY_MISSION = SELFLOOPING)
-#define DEFAULT_MAP_TRANSITION_CONFIG list(MAIN_STATION = CROSSLINKED, CENTCOMM = SELFLOOPING, EMPTY_AREA_1 = CROSSLINKED, EMPTY_AREA_2 = CROSSLINKED, MINING = SELFLOOPING, EMPTY_AREA_3 = CROSSLINKED, EMPTY_AREA_4 = CROSSLINKED, EMPTY_AREA_5 = CROSSLINKED, EMPTY_AREA_6 = CROSSLINKED, EMPTY_AREA_7 = CROSSLINKED, EMPTY_AREA_8 = CROSSLINKED)
+
+#define DECLARE_LEVEL(NAME,LINKS,TRAITS) list("name" = NAME, "linkage" = LINKS, "attributes" = TRAITS)
+
+
+#define DEFAULT_MAP_TRANSITION_CONFIG list(\
+DECLARE_LEVEL(MAIN_STATION,CROSSLINKED,list(STATION_LEVEL,STATION_CONTACT,REACHABLE,AI_OK)),\
+DECLARE_LEVEL(CENTCOMM,SELFLOOPING,list(ADMIN_LEVEL, BLOCK_TELEPORT, IMPEDES_MAGIC)),\
+DECLARE_LEVEL(EMPTY_AREA_1,CROSSLINKED,list(REACHABLE)),\
+DECLARE_LEVEL(EMPTY_AREA_2,CROSSLINKED,list(REACHABLE)),\
+DECLARE_LEVEL(MINING,SELFLOOPING,list(REACHABLE, STATION_CONTACT, HAS_WEATHER, ORE_LEVEL, AI_OK)),\
```

Comment:
We work by weird principles regarding that. We don't actually have mining. We will most likely need to adapt this heavily.